tensorboard/plugins/hparams/tf_hparams_google_analytics_tracker/tf-hparams-google-analytics-tracker.html

```diff
@@ -0,0 +1,261 @@
+     * @param {!CustomEvent} e The event to be processed.
+     */
+    handleEvent: function(e) {
+      if (!this.ga) {
+        console.warn('GA disabled: Would send', e.detail);
+        return;
+      }
+
+      this.ga('send', e.detail);
+    },
+
+    /**
+     * Send an "event" hit to Google Analytics.
+     *
+     * Due to the GA UI not easily displaying category, action and label
+     * together, it is recommended that you repeat the category in the action:
+     * c=Button, a=Click Button, l=Forward.
+     *
+     * @method sendEvent
+     * @param {string} category The event category, e.g. "Button".
+     * @param {string} action The type of interaction, e.g. "Click".
+     * @param {string=} opt_label Further categorization, e.g. "Forward".
+     * @param {number=} opt_value Non-negative. Useful to pass counts.
+     */
+    sendEvent: function(category, action, opt_label, opt_value) {
+      if (!this.ga) {
+        console.warn('GA disabled: Would send "event"',
+                     category, action, opt_label, opt_value);
+        return;
+      }
+
+      this.ga('send', 'event', category, action, opt_label, opt_value);
+    },
+
+    /**
+     * Send a "pageview" hit to Google Analytics.
+     *
+     * This also passes the document fragment identifier, unlike the default
+     * analytics.js API.
+     *
+     * @method sendPageView
+     */
+    sendPageView: function() {
+      var page = window.location.pathname +
+                 window.location.search +
+                 window.location.hash;
+
+      if (!this.ga) {
+        console.warn('GA disabled: Would send "pageview"', page);
+        return;
+      }
+
+      this.ga('send', 'pageview', {
+        'page': page
+      });
+    },
+
+    /**
+     * Send a "timing" hit to Google Analytics.
+     *
+     * Due to the GA UI not easily displaying category, varname and label
+     * together, it is recommended that you repeat the category in the varname:
+     * c=jQuery, v=Load jQuery, l=Google CDN.
+     *
+     * @method sendTiming
+     * @param {string} category A top-level categorization, e.g. "jQuery".
+     * @param {string} varname Name of the metric, e.g. "Load".
+     * @param {number} value The number of milliseconds of elapsed time.
+     * @param {string=} opt_label Further categorization, e.g. "Google CDN".
+     */
+    sendTiming: function(category, varname, value, opt_label) {
+      if (!this.ga) {
+        console.warn('GA disabled: would send "timing"',
+                     category, varname, value, opt_label);
+        return;
+      }
+
+      this.ga('send', 'timing', category, varname, value, opt_label);
+    }
+  });
+
+  /**
+   * Create a new "ga" object the way it's prescribed by Google Analytics.
+   *
+   * Copied from Google Analytics, but styling applied to make the JS linter
+   * happy.
+   *
+   * @return {!function(...)} a "ga" object proxy ready for use.
+   */
+  function newGa() {
+    (function(i, s, o, g, r, a, m) {
```

Comment:
Using analytics _within Google_ is fine, but we want to be very clear
that we’re not tracking our open source users at all. From looking at
this change as a whole, I can see that this tracking code isn’t actually
used, but that’s not locally obvious at all.

I would be much more comfortable if we could remove this tracking code
from the open-source version of TensorBoard entirely. TensorBoard
already does this in `tensorboard/components/analytics.html`:

<https://github.com/tensorflow/tensorboard/blob/382df2a414524effab9099c95190efa4ccfdd06a/tensorboard/components/analytics.html#L19-L24>

For the Google-internal version of TensorBoard, this file is swapped out
for a version that actually uses Google Analytics; in the open source
version, it’s obvious from the code that no tracking is or can be
performed.

Could you please either use or emulate this mechanism?

Reply:
Thanks Will. Sure. Let me do that and upload a new version.